### PR TITLE
Revert "Revert "take delius username from /users/{username}/details response instead of from login form input""

### PR DESCRIPTION
**Pull request by petergphillips**
Reverts ministryofjustice/hmpps-auth#532

### CODE REVIEW

File: src/test/groovy/uk/gov/justice/digital/hmpps/oauth2server/integration/specs/CommunityApiMockServer.groovy [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 import com.github.tomakehurst.wiremock.junit.WireMockRule
 
 class CommunityApiMockServer extends WireMockRule {
 
   CommunityApiMockServer() {
-    super(WireMockConfiguration.wireMockConfig().port(8099).usingFilesUnderClasspath("delius"))
+    super(WireMockConfiguration.wireMockConfig()
+        .port(8099)
+        .usingFilesUnderClasspath("delius")
+        .extensions(new ResponseTemplateTransformer(false)))
```
Comment (petergphillips) [inline src/test/groovy/uk/gov/justice/digital/hmpps/oauth2server/integration/specs/CommunityApiMockServer.groovy:13]: need to define the extension to get response templating working

File: src/test/groovy/uk/gov/justice/digital/hmpps/oauth2server/integration/specs/LoginSpecification.groovy [hunk 2]
```diff
@@ -79,21 +79,21 @@ class LoginSpecification extends DeliusIntegrationSpec {
         to LoginPage
 
         when: "I login with wrong password as a delius user"
-        loginAs DELIUS_LOCKED, 'wrongpassword'
+        loginAs 'DELIUS_LOCKED_IN_AUTH', 'wrongpassword'
```
Comment (petergphillips) [inline src/test/groovy/uk/gov/justice/digital/hmpps/oauth2server/integration/specs/LoginSpecification.groovy:82]: delius user locking actually happens in auth.  So need user that is known in delius and password gives a 401.  The `DELIUS_LOCKED` user is setup to be locked / disabled in delius so can't use that one

File: src/test/resources/delius/mappings/user_details_success.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "priority": 2,
   "request": {
     "method": "GET",
-    "urlPattern": "/secure/users/DELIUS.*/details"
+    "urlPattern": "(?i)/secure/users/DELIUS.*/details"
```
Comment (petergphillips) [inline src/test/resources/delius/mappings/user_details_success.json:5]: case insensitive matching of get user.  This matches how community api works.

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**